features/0586-sign-attachment/README.md

```diff
@@ -0,0 +1,219 @@
+- Status Note: A refinement of previous work
+- Start Date: 2020-08-18
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+- URI: http://didcomm.org/sign-attachment/%VER
+
+## Summary
+This protocol allows one agent to request another agent to sign an [attachment](https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0017-attachments) and return the signature.
+
+## Motivation
+
+The attachment is likely to be an agreement or a request, the signature of which applies agreement or approval by the signing party. The contents and structure of the attachment are outside the scope of this RFC, and should be detailed elsewhere.
+
+### Examples of application
+
+For Indy, in order to submit transactions to the ledger, it is often necessary for agents to obtain signatures from one or more endorsers. This protocol allows Authors to request signatures from Endorsers using DIDComm, and enable the Authors to create Indy transactions that can be written to the ledger.
+
+An agreement between two parties can be facilitated with this protocol. One party creates the agreement, and signs it. This protocol is used to get the signature from the other party, and the transaction described in the attachment is agreed to by both parties.
+
+## Tutorial
+
+### Name and Version
+
+Sign Attachment 1.0
+
+Also known as the "Please Sign This" protocol.
+
+URI: http://didcomm.org/sign-attachment/<version>/<messageType>
+
+## Key Concepts
+
+Attachment to sign - may be any format recognizable to all parties.
+
+Though this protocol can be used to gather signatures from multiple parties, each interaction is only between two parties. Multiple signatures are gathered by invoking the protocol once for each signer.
+
+```plantuml
+Coordinater --> Coordinater: Prepare Attachment
+Coordinater -> Signer1: Signature Request
+Signer1 -> Coordinater: Signature Response
+group Additional Signers
+    Coordinater -> SignerN: Signature Request
+    SignerN -> Coordinater: Signature Response
+end
+```
+
+## Roles
+
+**Coordinator** - Provides the document to sign, and coordinates with the signer(s).
+
+**Signer** - The party who's signature is requested. Evaluates the request from the coordinator, and either returns a signature as requested, or declines.
+
+## States
+
+**Requested** - The only state for both roles: This occurs after the signature_request has been sent/received, and before a response has been sent/received.
+
+```plantuml
+hide empty description
+[*] -> Requested: Request
+Requested -> [*] : Response (Signed)
+Requested -> [*] : Response (Declined)
+```
+## Messages
+
+### Signature Request
+
+This message represents a request from a "coordinator" to a "signer" to provide a signature for the attachment(s) included in the request.  The request will indicate the attached message type as well as the requested signature type, and "goal codes" will be included to indicate the "goal" of the coordinator, and stated goal of the signer.
```

Comment:
"the stated goal of the coordinator" for balance maybe?

I feel like maybe there should be only one goal - ie. the goal of the coordinator. The goal (or action) of the signer seems like it would always be implied by that goal.

Reply:
I've adjusted the language to reflect that this is from the opinion of the coordinator. The one goal discussion is a good one, but I didn't want to hold up the other changes for that discussion.